regress/unittests/sshbuf/test_sshbuf_getput_crypto.c

```diff
@@ -223,29 +240,74 @@ sshbuf_getput_crypto_tests(void)
 	TEST_DONE();
 
 #if defined(OPENSSL_HAS_ECC) && defined(OPENSSL_HAS_NISTP256)
-	TEST_START("sshbuf_put_ec");
+	TEST_START("sshbuf_put_ecpkey");
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+	param_bld = OSSL_PARAM_BLD_new();
+	ASSERT_PTR_NE(param_bld, NULL);
+	ASSERT_INT_EQ(OSSL_PARAM_BLD_push_utf8_string(param_bld,
+	    OSSL_PKEY_PARAM_GROUP_NAME, ec256_sn, strlen(ec256_sn)), 1);
+#else
 	eck = EC_KEY_new_by_curve_name(ec256_nid);
 	ASSERT_PTR_NE(eck, NULL);
 	ecp = EC_POINT_new(EC_KEY_get0_group(eck));
 	ASSERT_PTR_NE(ecp, NULL);
+#endif
 	MKBN(ec256_x, bn_x);
 	MKBN(ec256_y, bn_y);
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
```

Comment:
So (this comment applies to all of the bag of ifdefs like this)

This define should not be used in this code like this.  It precludes libressl from ever providing a compatible API
since both packages define OPENSSL_VERSION_NUMBER - Unless when providing 3.0 API you would prefer that we again jump the shark and sit on 0x40000000L or greater to get around this sort of ifdef mathemagics. 

again, I reiterate this is doing too much at once and should just concentrate on the compatible to 1.1.1. EVP_PKEY changes, but if a define for 3.0 is to be used, please make it a define for *OPENSSL 3.0*  not numeric magic like this.